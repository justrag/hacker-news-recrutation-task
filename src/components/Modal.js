import React from 'react';
import { connect } from 'react-redux';
import { IconButton, Content } from 'react-mdl';
import { getModalHref } from '../selectors/';
import { hideModal } from '../actions/';

const modalStyle = {
  position: 'fixed',
  backgroundColor: '#795548',
  zIndex: 1,
  left: '2%',
  top: '10%',
  width: '95%',
  height: '90%',
  //  overflow: 'auto',
  border: 'thin solid darkgrey'
};
const modalContentStyle = {
  backgroundColor: '#fefefe',
  margin: '15px auto',
  padding: '10px',
  border: '1px solid #888',
  width: '90%',
  height: '90%',
  display: 'block'
};

const closeButtonStyle = {
  float: 'right'
};

export const Modal = ({ href, hideModal }) =>
  href && (
    <Content style={modalStyle}>
      <IconButton
        style={closeButtonStyle}
        onClick={hideModal}
        name="close"
        colored
      />
      <iframe style={modalContentStyle} src={href} />
    </Content>
  );

const mapStateToProps = state => ({ href: getModalHref(state) });

export default connect(mapStateToProps, { hideModal })(Modal);
