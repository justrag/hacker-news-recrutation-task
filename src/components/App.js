import React from 'react';
import { Layout, Header, Footer } from 'react-mdl';
import Modal from './Modal';
import Stories from './Stories';

const layoutStyle = {
  background: 'url(Atari-800xl.jpg) center / cover'
};
//pic from http://bestandworstever.blogspot.com/2012/12/best-atari-system-ever-800xl.html

const titleStyle = { color: '#ddd' };
export const App = () => (
  <div>
    <Layout fixedHeader style={layoutStyle}>
      <Header title={<span style={titleStyle}>HackerNews</span>} />
      <Stories />
      <Modal />
      <Footer size="mini">
        <span>made by rag in 2018</span>
      </Footer>
    </Layout>
  </div>
);

export default App;
