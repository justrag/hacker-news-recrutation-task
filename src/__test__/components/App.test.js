import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../../components/App';

describe('Component: App', () => {
  it('renders correctly', () => {
    const test = shallow(<App />);
    expect(test).toMatchSnapshot();
  });
});
