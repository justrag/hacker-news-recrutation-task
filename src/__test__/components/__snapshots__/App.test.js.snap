// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App renders correctly 1`] = `
<div>
  <Layout
    fixedHeader={true}
    style={
      Object {
        "background": "url(Atari-800xl.jpg) center / cover",
      }
    }
  >
    <Header
      title={
        <span
          style={
            Object {
              "color": "#ddd",
            }
          }
        >
          HackerNews
        </span>
      }
    />
    <Connect(Component) />
    <Connect(Modal) />
    <Footer
      size="mini"
    >
      <span>
        made by rag in 2018
      </span>
    </Footer>
  </Layout>
</div>
`;
